### PR TITLE
Do not init database on witness nodes

### PR DESCRIPTION
**Pull request by vramesh7**
* Adds changes to make witness nodes not init rocks db.
* Add a Noninteractive client test that performs random get/put/delete operations.
* Add additional tests for replicated log logic.

### CODE REVIEW

File: paxos/replicated_log.cc [hunk 1]
```diff
@@ -200,13 +196,15 @@ ReplicatedLogEntry ReplicatedLog::GetLogEntryAtIdx(uint64_t idx) {
   absl::MutexLock l(&lock_);
   auto it = log_entries_.find(idx);
   CHECK(it != log_entries_.end());
+  it->second.idx_ = idx;
```
Comment (rsingh16) [inline paxos/replicated_log.cc:199]: I have both of this my PR, will remove it.

**Review by rsingh16 — APPROVED** [review]

File: client/client.cc [hunk 2]
```diff
@@ -80,87 +87,273 @@ Status KvsClient::DeleteGrpc(const std::string& key,
...
+        if (PutHelper(nodes, k, v) != 0) {
+          LOG(FATAL) << "[Client]: Put operation failed for test number: " << i;
+        } else {
+          LOG(INFO) << "[Client]: Put operation successful for: " << i;
+        }
+        kv[k] = v;
+        break;
+      }
+      case 2: {
+        std::uniform_int_distribution<> key_dis(0, static_cast<int>(i));
```
Comment (mmucklo) [inline client/client.cc:297]: do you just want the key to be in a range [0, i] on purpose?

File: paxos/paxos.cc [hunk 3]
```diff
@@ -51,6 +50,12 @@ PaxosResult Paxos::Propose(const std::string& value, uint8_t* leader_node_id,
     }
   }
 
+  // If the new leader is in the process of catching up, do not service
+  // reads/writes.
+  while (!this->paxos_node_->IsLeaderCaughtUp()) {
```
Comment (mmucklo) [inline paxos/paxos.cc:55]: Does this mean we're the leader here, or does this delay even happen on a replica?

**Review by mmucklo — APPROVED** [review]

File: client/client.cc (hunk 2 above)
Comment (vramesh7) [reply client/client.cc:297]: Yeah, I plan to follow this up with more tests.

File: paxos/paxos.cc (hunk 3 above)
Comment (vramesh7) [reply paxos/paxos.cc:55]: No, we should only reach here for leader

**Pull request merged**